rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -259,24 +274,10 @@ class Publisher : public PublisherBase
   }
 
 protected:
-  void
+  virtual void
   do_inter_process_publish(const MessageT & msg)
   {
-    auto status = rcl_publish(&publisher_handle_, &msg, nullptr);
-
-    if (RCL_RET_PUBLISHER_INVALID == status) {
-      rcl_reset_error();  // next call will reset error message if not context
-      if (rcl_publisher_is_valid_except_context(&publisher_handle_)) {
-        rcl_context_t * context = rcl_publisher_get_context(&publisher_handle_);
-        if (nullptr != context && !rcl_context_is_valid(context)) {
-          // publisher is invalid due to context being shutdown
-          return;
-        }
-      }
-    }
-    if (RCL_RET_OK != status) {
-      rclcpp::exceptions::throw_from_rcl_error(status, "failed to publish message");
-    }
+    generic_publisher::do_inter_process_publish(publisher_handle_, msg);
```

Comment:
I don't understand why the two new free functions are needed, and why a "normal" `Publisher` is calling the functions in `generic_publisher` namespace.
@DensoADAS can you expain?

Reply:
for the SerializedMessage the original "do_inter_process_publish" will fail as it's no ROS message

Reply:
can we just are an override for a serialized message in the `Publisher` class?